src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Models/RazorCodeActionExtensions.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using Newtonsoft.Json.Linq;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Models
+{
+    internal static class RazorCodeActionExtensions
+    {
+        public static CommandOrCodeAction AsVSCodeCommandOrCodeAction(this RazorCodeAction razorCodeAction)
+        {
+            if (razorCodeAction.Data is null)
+            {
+                // No command data
+                return new CommandOrCodeAction(new CodeAction()
```

Comment:
If there's no command data can't you just pass in the existing `RazorCodeAction` instead of creating a new one?

Reply:
`RazorCodeAction` hides the properties of O# `CodeAction`. I thought doing `CommandOrCodeAction(razorCodeAction)` would result in null properties as they were define don the `razorCodeAction` but not the O# properties.

Just tried it out and works fine, updated accordingly. Thanks!